### ISSUE TITLE
Question about Generator Update

### ISSUE THREAD

**Issue by artemishlee** [mit-han-lab/data-efficient-gans#48]
Hello, I tried to train a conditional SNGAN with and without DiffAugment on my own dataset, and got identical results when I ran inference. The FID and IS seem to be pretty close as well. 

Based on this, I think the gradients from DiffAugment are not being properly backpropagated. Would you have any suggestions on how to check for gradient flow or other ideas to see whether DiffAugment is working or not?

**Comment by zsyzzsoft** [comment]
How does your dataset look like? You may also check the training curves - with DiffAugment, G's loss should be lower and D's loss should be higher.

**Comment by artemishlee** [comment]
Thank you so much for the reply. I checked the training curves and indeed with DiffAugment, G's loss is lower and D's loss is higher compared to that without DiffAugment.

My dataset is synthetic - white background, with one object in it with 3-4 different colors to represent different parts of the object. 

I'm still not quite sure why there seems to be no improvement with DiffAugment. Somehow the training curves are different but the gradient update seems to be the same for the generator.

**Comment by zsyzzsoft** [comment]
Maybe the dataset is simple but large enough to prevent the discriminator from overfitting. If you are interested, you may try DiffAugment when reducing the number of training samples.

**Comment by artemishlee** [comment]
Thanks, I will try that. I have about 1K images or so, so I didn't think the dataset was large (but maybe the simplicity compensates for that). The FID isn't very good though (~50).

**Comment by artemishlee** [comment]
It seems like training with 10% of the data with and without DiffAugment isn't making much difference either. :( I think I saw from another issue that the range of the DiffAugmented images may need to be something like [0,1] in order for gradients to flow?

My images before DiffAugment ranges from [-1, 1], then after DiffAugment (color, translation, and cutout) it is around [-2, 2].

**Comment by zsyzzsoft** [comment]
[-1, 1] should be fine. Could you share some generated images with and without DiffAugment?
